ee/desktop/notify/notify_darwin.m

```diff
@@ -1,16 +1,46 @@
 //go:build darwin
 // +build darwin
 
-// Draws from Fyne's implementation: https://github.com/fyne-io/fyne/blob/master/app/app_darwin.m
+// sendNotification draws from Fyne's implementation: https://github.com/fyne-io/fyne/blob/master/app/app_darwin.m
 
-#import <Foundation/Foundation.h>
-#import <UserNotifications/UserNotifications.h>
+#import "notify_darwin.h"
 
-BOOL doSendNotification(UNUserNotificationCenter *center, NSString *title, NSString *body) {
+@implementation NotificationDelegate
+- (void)userNotificationCenter:(UNUserNotificationCenter *)center didReceiveNotificationResponse:(UNNotificationResponse *)response withCompletionHandler:(void (^)(void))completionHandler {
+    NSDictionary *userInfo = response.notification.request.content.userInfo;
+
+    NSString *actionUri = userInfo[@"action_uri"];
+    if ([actionUri length] != 0) {
+        [[NSWorkspace sharedWorkspace] openURL:[NSURL URLWithString:actionUri]];
+    }
+
+    completionHandler();
+}
+@end
+
+void runNotificationListenerApp(void) {
+    [NSAutoreleasePool new];
```

Comment:
I wonder if this whole function could just be wrapped like:

```
@autoreleasepool {
...
}
```

Then everything gets released automatically.

Reply:
I'll give it a shot!

Reply:
Looks like it works 🙂